File: go/sieve/sieve.go
package sieve

const testVersion = 1

func Sieve(limit int) (primes []int) {
	s := map[int]bool{}
	for i := 2; i < limit; i++ {
		for j := 2; ; j++ {
			if i*j > limit {
				break
			} else {
				s[i*j] = true
			}
		}
	}
	for i := 2; i <= limit; i++ {
		if !s[i] {
			primes = append(primes, i)
		}
	}
	return primes
}

File: python/sieve/sieve.py
def sieve(limit):
    if limit == 1:
        return []
    s = [True] * (limit + 1)
    for i in range(2, limit + 1):
        if not s[i]:
            continue
        for j in range(2, (limit // i) + 1):
            s[i * j] = False
    return [i for i in range(2, limit + 1) if s[i]]
